fem/kernels.hpp

```diff
@@ -26,6 +26,50 @@ namespace kernels
 namespace internal
 {
 
+/// Load B1d matrice into shared memory
+template<int MD1, int MQ1>
+MFEM_HOST_DEVICE inline void LoadB(const int D1D, const int Q1D,
+                                   const ConstDeviceMatrix b,
```

Comment:
Pass by reference, not by value.

This also applies to other similar function parameters below.

Reply:
`done`